packages/jest-runtime/src/__tests__/Runtime-requireModule-test.js

```diff
@@ -274,4 +274,24 @@ describe('Runtime requireModule', () => {
     ]),
   );
 
+  it('finds modules encoded in UTF-8 *with BOM*', () =>
+    createRuntime(__filename).then(runtime => {
+      const exports = runtime.requireModule(
+        runtime.__mockRootPath,
+        './UTF8WithBOM.js',
+      );
+      expect(exports).toBe('isModuleEncodedInUTF8WithBOM');
+    }),
+  );
+
+  it('finds and loads JSON files encoded in UTF-8 *with BOM*', () =>
+    createRuntime(__filename).then(runtime => {
+      const exports = runtime.requireModule(
+        runtime.__mockRootPath,
+        './UTF8WithBOM.json',
+      );
+      expect(exports.isJSONModuleEncodedInUTF8WithBOM).toBe(true);
+    }),
+  );
```

Comment:
@cpojer Whilst the JSON test will fail in master the JS test will actually pass on master as it seems although 'vm' actually handles BOMs but I figured it's still useful to test for in case things change in Node. -- my guess is ['module' needed to strip it](https://github.com/nodejs/node/blob/master/lib/module.js#L578) because of the old implementation of 'vm' in Node...